api/lib/infrastructure/repositories/jury-session-repository.js

```diff
@@ -0,0 +1,84 @@
+      .select(Bookshelf.knex.raw('COUNT(*) FILTER (WHERE ?? IS TRUE) OVER (PARTITION BY ??) AS ??',
+        ['certification-courses.isPublished', 'sessions.id', 'countPublishedCertifications']))
+      .from('sessions')
+      .leftJoin('certification-centers', 'certification-centers.id', 'sessions.certificationCenterId')
+      .leftJoin('certification-courses', 'certification-courses.sessionId', 'sessions.id')
+      .leftJoin('users', 'users.id', 'sessions.assignedCertificationOfficerId')
+      .where('sessions.id', '=', id)
+      .limit(1);
+    if (!results[0]) {
+      throw new NotFoundError('La session n\'existe pas ou son accès est restreint');
+    }
+    return _toDomain(results[0]);
+  },
+
+  async findPaginatedFiltered({ filters, page }) {
+    const pageSize = page.size ? page.size : 10;
+    const pageNumber = page.number ? page.number : 1;
+    const offset = (pageNumber - 1) * pageSize;
+    const results = await Bookshelf.knex.with('the_sessions', (qb) => {
+      const { id } = filters;
+      qb.select('*')
+        .select(Bookshelf.knex.raw('COUNT(*) OVER() AS ??', ['rowCount']))
+        .from('sessions');
+      if (id) {
+        qb.where({ id });
+      }
+      qb.orderByRaw('?? ASC NULLS FIRST', 'publishedAt');
+      qb.orderByRaw('?? ASC', 'finalizedAt');
+      qb.limit(pageSize).offset(offset);
+    })
+      .distinct('the_sessions.*')
+      .select('certification-centers.type', 'users.firstName', 'users.lastName')
+      .select(Bookshelf.knex.raw('COUNT(*) FILTER (WHERE ?? IS TRUE) OVER (PARTITION BY ??) AS ??',
+        ['certification-courses.isPublished', 'the_sessions.id', 'countPublishedCertifications']))
+      .from('the_sessions')
+      .leftJoin('certification-centers', 'certification-centers.id', 'the_sessions.certificationCenterId')
+      .leftJoin('certification-courses', 'certification-courses.sessionId', 'the_sessions.id')
+      .leftJoin('users', 'users.id', 'the_sessions.assignedCertificationOfficerId');
+
+    let rowCount = 0;
+    const jurySessions = _.map(results, (result) => {
+      rowCount = result.rowCount;
+      return _toDomain(result);
+    });
+
+    return {
+      jurySessions,
+      pagination: {
+        page: pageNumber,
+        pageSize: pageSize,
+        rowCount,
+        pageCount: Math.ceil(rowCount / pageSize),
+      },
+    };
+  },
+};
+
+function _toDomain(jurySessionFromDB) {
+  let assignedCertificationOfficer;
+  if (jurySessionFromDB.assignedCertificationOfficerId) {
+    assignedCertificationOfficer = new CertificationOfficer({
+      id: jurySessionFromDB.assignedCertificationOfficerId,
+      firstName: jurySessionFromDB.firstName,
+      lastName: jurySessionFromDB.lastName,
+    });
+  }
+  return new JurySession({
+    ...jurySessionFromDB,
+    certificationCenterName: jurySessionFromDB.certificationCenter,
+    certificationCenterType: jurySessionFromDB.type,
+    assignedCertificationOfficer,
+  });
```

Comment:
Dans le mesure où on a une partie requise et un attribut optionnel, pour simplifier la lectur eet la maintenance, je propose de révéler cet état de fait :

```suggestion
  const jurySession = new JurySession({
    ...jurySessionFromDB,
    certificationCenterName: jurySessionFromDB.certificationCenter,
    certificationCenterType: jurySessionFromDB.type,
  });

  if (jurySessionFromDB.assignedCertificationOfficerId) {
    const assignedCertificationOfficer = new CertificationOfficer({
      id: jurySessionFromDB.assignedCertificationOfficerId,
      firstName: jurySessionFromDB.firstName,
      lastName: jurySessionFromDB.lastName,
    });
    jurySession.assignedCertificationOfficer = assignedCertificationOfficer;
  }
  
  return jurySession;
```

Reply:
Oui bonne idée, done au prochain push